clayui.com/src/components/IconSearch/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useEffect, useState} from 'react';
+
+const spritemap = '/images/icons/icons.svg';
+
+const IconSearch = source => {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [filteredIconNames, setFilteredIconNames] = useState([]);
+	const list = filteredIconNames.length ? filteredIconNames : [];
+
+	source = [...source.source];
+
+	useEffect(() => {
+		const filteredIcons = source.filter(icon => {
+			const iconPrimary = icon[Object.keys(icon)[0]].toUpperCase();
```

Comment:
Instead of using `Object.keys(icon)` and picking indexes, if we have a strict type for the icons object, we can just get the items by its key. This would require us to define the flags and icons json to be the same schema, which should be fine since we have control of that. This would allow us to clean up the filtering logic. For example...

```ts
// Imagine this is the type of the icons provided
type iconsSourceJSON = Array<{
	aliases: Array<string>;
	name: string;
}>;

// Logic for filtering
// `source` is the iconsSourceJSON type
const filteredIcons = source.filter(icon => {
	const allItems = [icon.name, ...icon.aliases].map(item =>
		item.toUpperCase()
	);

	return allItems.includes(searchQuery.toUpperCase());
});
```

Reply:
Yeah that was the original implementation but I went with a more vague approach which made the code uglier. 
I prefer having them follow the same schema, not sure how to enforce or inform future devs of the component that their source JSON needs to have a specific schema.

Reply:
You could add a comment jsdoc block at the top of the component and define props and type accepted.